domain/consensus/processes/blockprocessor/validateandinsertblockErrorCases_test.go

```diff
@@ -0,0 +1,126 @@
+package blockprocessor_test
+
+import (
+	"github.com/kaspanet/kaspad/domain/consensus"
+	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/domain/consensus/ruleerrors"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/testutils"
+	"github.com/kaspanet/kaspad/domain/dagconfig"
+	"github.com/pkg/errors"
+	"strings"
+	"testing"
+)
+
+// All the tests below are tests the error cases in "validateAndInsertBlock" function.
+// Each test is coverage the error cases in sub-function in "validateAndInsertBlock" function.
+// Currently, implement only for some of the errors.
+
+// TestErrorCasesOnValidateBlock tests the error cases on "validateBlock" function.
+func TestErrorCasesOnValidateBlock(t *testing.T) {
+	testutils.ForAllNets(t, true, func(t *testing.T, params *dagconfig.Params) {
+		factory := consensus.NewFactory()
+		tc, teardown, err := factory.NewTestConsensus(params, false, "TestBlockStatus")
+		if err != nil {
+			t.Fatalf("Error setting up consensus: %+v", err)
+		}
+		defer teardown(false)
+		tipHash, emptyCoinbase, tx1 := initData(params)
+
+		// Tests all the error case on the function: "checkBlockStatus"(sub-function in function validateBlock)
+		blockWithStatusInvalid, err := tc.BuildBlock(&emptyCoinbase, []*externalapi.DomainTransaction{tx1, tx1})
+		if err != nil {
+			t.Fatalf("AddBlock: %+v", err)
+		}
+		_, err = tc.ValidateAndInsertBlock(blockWithStatusInvalid)
+		if err == nil {
+			t.Fatalf("Test ValidateAndInsertBlock: Expected an error, because the block is invalid.")
+		}
+		_, err = tc.ValidateAndInsertBlock(blockWithStatusInvalid)
+		if err == nil {
+			t.Fatalf("Test ValidateAndInsertBlock: Expected an error, because the block is invalid.")
+		}
+		if !errors.Is(err, ruleerrors.ErrKnownInvalid) {
+			t.Fatalf("Expected block to be invalid with err: %v, instead found: %v", ruleerrors.ErrKnownInvalid, err)
+		}
```

Comment:
Do you insert the same block twice on purpose or is that by mistake?
also, IMO there's no need to check both that the err isn't nil and that it's the correct error, either write `err == nil || !errors.Is(err, SomeErr)` or just write `!errors.Is(err, SomeErr)` directly

Reply:
I inserted the same block twice because the error is triggered only if the block already exists.
I'll fix the second comment.